starlette/exceptions.py

```diff
@@ -23,6 +24,15 @@ def __repr__(self) -> str:
         return f"{class_name}(status_code={self.status_code!r}, detail={self.detail!r})"
 
 
+class WebSocketException(Exception):
+    def __init__(self, code: int) -> None:
```

Comment:
Since [`reason` has been implemented](https://github.com/encode/starlette/issues/991), shouldn't we consider adding an optional reason field for the class as well?

Reply:
yep, thanks